src/Object/Contact.php

```diff
@@ -21,29 +21,30 @@
 require_once 'boot.php';
 require_once 'include/text.php';
 
+/**
+ * @brief functions for interacting with a contact
+ */
 class Contact extends BaseObject
 {
 	/**
 	 * @brief Marks a contact for removal
 	 *
-	 * @param int $id
+	 * @param int $id contact id
 	 * @return null
 	 */
 	public static function remove($id)
 	{
 		// We want just to make sure that we don't delete our "self" contact
-		$r = q(
-			"SELECT `uid` FROM `contact` WHERE `id` = %d AND NOT `self` LIMIT 1", intval($id)
-		);
+		$condition = array('`id` = ? AND NOT `self`', $id);
+		$r = dba::select('contact', array('uid'), $condition, array('limit' => 1));
```

Comment:
When you call `dba::select()` with `limit => 1`, it returns the actual row. You don't need `$r[0]` afterwards.